bindings/java/src/org/sleuthkit/datamodel/AbstractContent.java

```diff
@@ -318,42 +319,60 @@ public int hashCode() {
 		return hash;
 	}
 
+	@Deprecated
 	@Override
 	public BlackboardArtifact newArtifact(int artifactTypeID) throws TskCoreException {
 		// don't let them make more than 1 GEN_INFO
 		if (artifactTypeID == ARTIFACT_TYPE.TSK_GEN_INFO.getTypeID()) {
 			return getGenInfoArtifact(true);
 		}
-		return db.newBlackboardArtifact(artifactTypeID, objId);
+		BlackboardArtifact.Type artifactType = db.getArtifactType(artifactTypeID);
```

Comment:
We should make SleuthkitCase.getArtifactType() do a lookup in the in the standard types map before doing a query. I will write a story about this.

Reply:
It looks like SleuthkitCase.getArtifactType() has caching and gets initialized with all the standard types